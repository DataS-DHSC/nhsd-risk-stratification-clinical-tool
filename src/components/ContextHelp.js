import PropTypes from 'prop-types';
import { ROUTE_CLINICAL_GUIDANCE } from '../constants/routes';

function ContextHelp(props) {
  const { helpPageReference, labelText } = props;
  return (
    <>
      {/* eslint-disable-next-line react/jsx-no-target-blank */}
      <a
        href={`${ROUTE_CLINICAL_GUIDANCE}#${helpPageReference}`}
        rel="opener"
        target="_blank"
      >
        {labelText}
      </a>
    </>
  );
}

ContextHelp.propTypes = {
  helpPageReference: PropTypes.string,
  labelText: PropTypes.string.isRequired,
};

ContextHelp.defaultProps = {
  helpPageReference: 'top',
};

export default ContextHelp;
