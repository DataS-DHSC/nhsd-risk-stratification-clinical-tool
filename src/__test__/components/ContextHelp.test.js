import React from 'react';
import { render } from '@testing-library/react';
import ContextHelp from '../../components/ContextHelp';

const setup = (props) => render(<ContextHelp {...props} />);

describe('ContextHelp', () => {
  it('should match snapshot for top link', () => {
    const container = setup({ labelText: 'More help is available' });
    expect(container.asFragment()).toMatchSnapshot();
  });

  it('should match snapshot for title link', () => {
    const container = setup({
      helpPageReference: 'patient-details',
      labelText: 'Patient details help',
    });
    expect(container.asFragment()).toMatchSnapshot();
  });
});
