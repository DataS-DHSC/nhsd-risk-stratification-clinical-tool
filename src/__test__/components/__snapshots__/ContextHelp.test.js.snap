// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextHelp should match snapshot for title link 1`] = `
<DocumentFragment>
  <a
    href="https://digital.nhs.uk/coronavirus/risk-assessment/clinical-tool/guidance-for-clinicians#patient-details"
    rel="opener"
    target="_blank"
  >
    Patient details help
  </a>
</DocumentFragment>
`;

exports[`ContextHelp should match snapshot for top link 1`] = `
<DocumentFragment>
  <a
    href="https://digital.nhs.uk/coronavirus/risk-assessment/clinical-tool/guidance-for-clinicians#top"
    rel="opener"
    target="_blank"
  >
    More help is available
  </a>
</DocumentFragment>
`;
